src/components/SelectAddressDialog/index.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withStyles } from 'material-ui/styles';
+import Button from 'material-ui/Button';
+import List, { ListItem, ListItemAvatar, ListItemText } from 'material-ui/List';
+import Dialog, { DialogTitle } from 'material-ui/Dialog';
+import Typography from 'material-ui/Typography';
+
+import styles from './styles';
+
+class SelectAddressDialog extends React.Component {
+  static propTypes = {
+    classes: PropTypes.object.isRequired,
+    dialogVisible: PropTypes.bool.isRequired,
+    walletAddresses: PropTypes.array.isRequired,
+    onClosed: PropTypes.func.isRequired,
+  };
+
+  render() {
+    const {
+      classes,
+      dialogVisible,
+      walletAddresses,
+    } = this.props;
+
+    return (
+      <Dialog
+        open={dialogVisible}
+        onClose={this.handleClose}
+      >
+        <DialogTitle>Select Centralized Oracle</DialogTitle>
```

Comment:
let's call it result setter and please handle the localization

Reply:
done